### PR TITLE
Implemented the ARM SEL instruction, with tests.

### PR DESCRIPTION
**Pull request by Subv**
The test for this instruction is very peculiar. As the instruction's behavior depends on the value of the CPSR, we generate a MSR instruction after each SEL instruction to change the CPSR.

### CODE REVIEW

File: src/frontend/translate/translate_arm/status_register_access.cpp [hunk 1]
```diff
@@ -77,6 +77,42 @@ bool ArmTranslatorVisitor::arm_SRS() {
     return InterpretThisInstruction();
 }
 
+bool ArmTranslatorVisitor::arm_SEL(Cond cond, Reg n, Reg d, Reg m) {
+    if (n == Reg::PC || d == Reg::PC || m == Reg::PC)
+        return UnpredictableInstruction();
+
+    if (ConditionPassed(cond)) {
+
```
Comment (lioncash) [inline src/frontend/translate/translate_arm/status_register_access.cpp]: Unnecessary newline.

File: src/frontend/translate/translate_arm/status_register_access.cpp [hunk 2]
```diff
@@ -77,6 +77,42 @@ bool ArmTranslatorVisitor::arm_SRS() {
...
+        auto cpsr = ir.GetCpsr();
+
+        auto to = ir.GetRegister(m);
+        auto from = ir.GetRegister(n);
+
+        auto one = ir.Imm32(1);
+        auto zero = ir.Imm1(0);
```
Comment (lioncash) [inline src/frontend/translate/translate_arm/status_register_access.cpp]: Imm1 takes a boolean value, not an integral one.

**Comment by Subv** [comment]
Updated with the algorithm provided by @yuriks to reduce the number of instructions needed when expanding the GE bits.

File: src/frontend/translate/translate_arm/status_register_access.cpp [hunk 3]
```diff
@@ -77,6 +77,40 @@ bool ArmTranslatorVisitor::arm_SRS() {
...
+        auto from = ir.GetRegister(n);
+
+        auto zero = ir.Imm1(false);
+
+        // Extract bits 16:19 from the CPSR
+        auto bits = ir.And(ir.LogicalShiftRight(cpsr, ir.Imm8(16), zero).result, ir.Imm32(0xF));
+
+        // Perform some arithmetic to expand 0bXYZW into 0bXXXXXXXXYYYYYYYYZZZZZZZZWWWWWWWW => 0xXXYYZZWW
+        // The logic behind this is as follows:
+        /*
```
Comment (lioncash) [inline src/frontend/translate/translate_arm/status_register_access.cpp]: Please use single-line comments

**Comment by Subv** [comment]
Addressed

**Comment by yuriks** [comment]
Oops, there's a better way, even:
```
0000 0000 0000 0000|0000 0000 0000 xyzw
0000 000x yzw0 00xy|zw00 0xyz w000 xyzw  (x * 0x00204081)
0000 000x 0000 000y|0000 000z 0000 000w  (x & 0x01010101)
xxxx xxxx yyyy yyyy|zzzz zzzz wwww wwww  (x * 0xff)
```

**Comment by Subv** [comment]
@yuriks Done.

**Comment by MerryMage** [comment]
LGTM

**Pull request merged**